animation_nodes/nodes/color/set_vertex_color.py

```diff
@@ -1,62 +1,100 @@
 import bpy
-import numpy
 from bpy.props import *
 from ... events import propertyChanged
-from ... base_types import AnimationNode
+from ... base_types import AnimationNode, VectorizedSocket
+from ... data_structures import Color, ColorList, VirtualColorList
+from . c_utils import getLoopColorsFromVertexColors, getLoopColorsFromPolygonColors
+
+colorModeItems = [
+    ("LOOP", "Loop", "Set color of every loop vertex", "NONE", 0),
+    ("VERTEX", "Vertex", "Set color of every vertex", "NONE", 1),
+    ("POLYGON", "Polygon", "Set color of every polgon", "NONE", 2)    
+]
+
+colorLayerIdentifierTypeItems = [
+    ("INDEX", "Index", "Get vertex color layer based on the index", "NONE", 0),
+    ("NAME", "Name", "Get vertex color layer based on the name", "NONE", 1)
+]
 
 class SetVertexColorNode(bpy.types.Node, AnimationNode):
     bl_idname = "an_SetVertexColorNode"
     bl_label = "Set Vertex Color"
     errorHandlingType = "EXCEPTION"
 
-    vertexColorName: StringProperty(name = "Vertex Color Group", default = "Col", update = propertyChanged)
-    checkIfColorIsSet: BoolProperty(default = True)
+    colorMode: EnumProperty(name = "Color Mode", default = "LOOP",
+        items = colorModeItems, update = AnimationNode.refresh)
+
+    colorLayerIdentifierType: EnumProperty(name = "Color Layer Identifier Type", default = "INDEX",
+        items = colorLayerIdentifierTypeItems, update = AnimationNode.refresh)
+
+    useColorList: VectorizedSocket.newProperty()
 
     def create(self):
-        self.newInput("Object", "Object", "object").defaultDrawType = "PROPERTY_ONLY"
-        self.newInput("Color", "Color", "color").defaultDrawType = "PROPERTY_ONLY"
-        self.newOutput("Object", "Object", "outObject")
+        self.newInput("Object", "Object", "object", defaultDrawType = "PROPERTY_ONLY")
 
-    def draw(self, layout):
-        layout.prop(self, "vertexColorName", text = "", icon = "GROUP_VCOL")
+        if self.colorLayerIdentifierType == "INDEX":
+            self.newInput("Integer", "Color Index", "colorLayerIndex")
+        elif self.colorLayerIdentifierType == "NAME":
+            self.newInput("Text", "Name", "colorLayerName")
 
-    def drawAdvanced(self, layout):
-        layout.prop(self, "checkIfColorIsSet", text = "Check Color")
+        self.newInput(VectorizedSocket("Color", "useColorList",
+            ("Color", "color"), ("Colors", "colors")))
 
-    def execute(self, object, color):
-        if object is None: return object
-        if object.type != "MESH": return object
-        if object.mode == "EDIT":
-            self.raiseErrorMessage("Object is in edit mode")
-            return object
+        self.newOutput("Object", "Object", "outObject")
 
-        mesh = object.data
+    def draw(self, layout):
+        layout.prop(self, "colorMode", text = "")
```

Comment:
Only draw the enum if `useColorList` is true.

Reply:
It is added.